### DISCUSSION TITLE
Preserve state in child compontent of Drawer

### DISCUSSION THREAD

**Discussion by mariansimecek** [mantinedev/mantine#3292] [Q&A] [answered]
Hi, is there some easy way to preserve the state inside **ChildComponent** placed in **Drawer**? Or I just must lift up state?

```tsx
    <Drawer
        ...
      >
        <ChildComponent/>
      </Drawer>
```

Thanks for answer

**Comment by rtivital** [comment, marked as answer]
Lift the state up

**Reply by mariansimecek** [reply]
thank you

**Reply by mariansimecek** [reply]
@rtivital It would be nice to have some prop that will not remove children from DOM but instead hide with CSS. Something that has mui-x library with keepMounted prop. But don't know if is a good idea or not.

**Reply by rtivital** [reply]
I'll see if it is possible
